db/migrate/20151102211125_add_slug_to_bit_core_tool.rb

```diff
@@ -0,0 +1,5 @@
+class AddSlugToBitCoreTool < ActiveRecord::Migration
+  def change
+    add_column :bit_core_tools, :slug, :string
```

Comment:
Nope.  Having it `nil` is fine.  If it is `nil` then it will just reference the `title`.  And title has a constraint on it.